src/core_atmosphere/mpas_atm_core.F

```diff
@@ -295,6 +296,12 @@ function atm_core_init(domain, startTimeStamp) result(ierr)
       !
       call mpas_atm_dynamics_init(domain)
 
+      !
+      ! Initialize the chemistry package
+      ! Assume for now that the chemistry solver will work on columns
+      !
+      call chemistry_init(nVertLevels)
```

Comment:
Rather than passing a specific (vertical) dimension here for the initialization of all chemistry, I think it would be better to pass some higher-level data structures, perhaps the `configs` and `dimensions` pools (which can be given here as `domain % blocklist % configs` and `domain % blocklist % dimensions`). Then, within the `chemistry_init` routine, the musica initialization routine can get just `nVertLevels` if it chooses, while other chem/aerosol packages like GOCART-2G can obtain namelist options and/or dimensions.

Reply:
In mpas_atm_core.F, changed to `chemistry_init(domain % blocklist % configs, domain % blocklist % dimensions)`.

Reply:
It seems once for the entire model.
It looks like `domain % blocklist` is initialized by `mpas_atm_threading_init`.
My understanding is that each MPI rank can be assigned multiple blocks (`config_number_of_blocks`)
```
atm_core_init(domain, startTimeStamp)
    mpas_atm_threading_init(domain % blocklist, ierr)
    ...
    chemistry_init(domain % blocklist % configs, domain % blocklist % dimensions)
```